src/components/SurveyComponents/GeneralQuestions.tsx

```diff
@@ -0,0 +1,336 @@
+  },
+  level: {
+    type: 'MCQ',
+    question: 'Your level at your current company is:',
+    choices: Object.values(levelMap),
+    isRequired: true,
+  },
+  yearsOfExperience: {
+    type: 'number',
+    question: 'Years of experience:',
+    isRequired: true,
+  },
+  companiesLast5Years: {
+    type: 'number',
+    question: 'How many companies did you work at in the last 5 years?',
+    isRequired: true,
+  },
+  sameCompanyLastYear: {
+    type: 'MCQ',
+    question: 'Did you work at the same company last year (January 2023) ?',
+    choices: ['Yes', 'No'],
+    isRequired: false,
+  },
+  workLocation: {
+    type: 'MCQ',
+    question: 'You work from',
+    choices: [
+      'Office for a company in Egypt',
+      'Home for a company in Egypt',
+      'Hybrid for a company in Egypt (e.g 2-3 days from home or office)',
+      'Home for a company abroad (Remote employee)',
+      'Abroad in another country (Relocated elsewhere)',
+    ],
+    isRequired: true,
+  },
+  companySize: {
+    type: 'MCQ',
+    question: 'Your company is considered:',
+    choices: [
+      'A Start-up',
+      'A Small and Medium Enterprise',
+      'A Large Enterprise/Multinational Company',
+    ],
+    isRequired: true,
+  },
+  companyLocationEgypt: {
+    type: 'MCQ',
+    question: "Company Location in case it's operating from Egypt:",
+    choices: [
+      'No site in Egypt',
+      'Cairo',
+      'Alexandria',
+      'Hurghada (Or Gouna)',
+      'Port Said / Suez',
+      'Mansoura',
+      'Faiyum',
+      'Luxor / Aswan',
+      'Other',
+    ],
+    isRequired: true,
+  },
+  companyType: {
+    type: 'MCQ',
+    question: 'You work in a:',
+    choices: ['Software House', 'Product-based company', 'Other'],
+    isRequired: true,
+  },
+  businessFocus: {
+    type: 'MCQ',
+    question: 'The business main focus is:',
+    choices: Object.values(businessLineMap),
+    isRequired: true,
+  },
+  businessScope: {
+    type: 'MCQ',
+    question: 'The business is',
+    choices: [
+      'Local / National (operates inside Egypt)',
+      'Regional (has some business lines in Saudi Arabia, Emirates, etc)',
+      'Global (has some business lines in the region, the US, Europe, Asia, etc.)',
+    ],
+    isRequired: false,
+  },
+  businessIndustries: {
+    type: 'checkbox',
+    question: 'The industries of the business are',
+    choices: [
+      'Fintech',
+      'Healthcare',
+      'Insurance',
+      'Automotive',
+      'Delivery',
+      'Supply Chain',
+      'Transportation',
+      'Telecommunications',
+      'Energy',
+      'Manufacturing',
+      'Technology',
+      'Education',
+      'Human Resources Services',
+      'Law',
+      'Agriculture',
+      'Production',
+      'Aerospace',
+      'Other',
+    ],
+    isRequired: false,
+  },
+}
+
+const engineeringRoles = new Set(
+  POSITIONS_OPTIONS.filter((position) =>
+    ['engineer', 'developer', 'data'].includes(position.category),
+  ).map((pos) => pos.positionName),
```

Comment:
since you are making a set out of filtered then mapped array items a lot, you can extract this functionality into a util function for reusability